File: 武将评级划分/data/process_treasure_tier.py
# -*- coding: utf-8 -*-
import pandas as pd
import json
import os

# 设置路径
data_dir = r"c:\Users\zhangfan\my-openspec-project\武将评级划分\data"
output_path = os.path.join(data_dir, "treasure_tier_result.json")

print("=" * 60)
print("读取Excel文件...")
print("=" * 60)

# 读取Excel文件
general_conf = pd.read_excel(os.path.join(data_dir, "sgs_general_conf.xlsx"))
item_general_conf = pd.read_excel(os.path.join(data_dir, "sgs_item_general_conf.xlsx"))
price_table = pd.read_excel(os.path.join(data_dir, "新三国杀道具定价表.xlsx"))

print("\n=== sgs_general_conf.xlsx (武将配置表) ===")
print("Columns:", general_conf.columns.tolist())
print("Shape:", general_conf.shape)
print(general_conf.head(3).to_string())

print("\n=== sgs_item_general_conf.xlsx (武将道具表) ===")
print("Columns:", item_general_conf.columns.tolist())
print("Shape:", item_general_conf.shape)
print(item_general_conf.head(3).to_string())

print("\n=== 新三国杀道具定价表.xlsx (道具定价表) ===")
print("Columns:", price_table.columns.tolist())
print("Shape:", price_table.shape)
print(price_table.head(10).to_string())

# 查找GetValue列
print("\n" + "=" * 60)
print("分析GetValue和珍宝价格关联...")
print("=" * 60)

# 找到包含GetValue的列
getvalue_cols = [col for col in general_conf.columns if 'GetValue' in str(col)]
print(f"\nGetValue相关列: {getvalue_cols}")

# 找到包含珍宝价格的列
price_cols = [col for col in price_table.columns if any(kw in str(col) for kw in ['珍宝', '价格', 'price', 'Price'])]
print(f"价格相关列: {price_cols}")

# 显示道具表的关键列
item_cols = [col for col in item_general_conf.columns if any(kw in str(col) for kw in ['general', 'item', 'value', 'type', 'id'])]
print(f"道具表关键列: {item_cols}")

# 显示道具表内容
print("\n道具表内容预览:")
print(item_general_conf.head(10).to_string())

# 显示定价表内容
print("\n定价表内容预览:")
print(price_table.head(10).to_string())

File: 武将评级划分/data/process_treasure_tier.js
// -*- coding: utf-8 -*-
const XLSX = require('xlsx');
const path = require('path');
const fs = require('fs');

const dataDir = path.join(__dirname);
const outputPath = path.join(dataDir, 'treasure_tier_result.json');

console.log('='.repeat(60));
console.log('读取Excel文件...');
console.log('='.repeat(60));

// 读取Excel文件
const generalConf = XLSX.readFile(path.join(dataDir, 'sgs_general_conf.xlsx'));
const itemGeneralConf = XLSX.readFile(path.join(dataDir, 'sgs_item_general_conf.xlsx'));
const priceTable = XLSX.readFile(path.join(dataDir, '新三国杀道具定价表.xlsx'));

// 转换为sheet数据
function sheetToJson(workbook) {
    const sheetName = workbook.SheetNames[0];
    return XLSX.utils.sheet_to_json(workbook.Sheets[sheetName]);
}

const generals = sheetToJson(generalConf);
const items = sheetToJson(itemGeneralConf);
const prices = sheetToJson(priceTable);

console.log('\n=== sgs_general_conf.xlsx (武将配置表) ===');
console.log('列名:', Object.keys(generals[0] || {}));
console.log('数据量:', generals.length);
console.log('示例数据:', JSON.stringify(generals[0], null, 2));

console.log('\n=== sgs_item_general_conf.xlsx (武将道具表) ===');
console.log('列名:', Object.keys(items[0] || {}));
console.log('数据量:', items.length);
console.log('示例数据:', JSON.stringify(items[0], null, 2));

console.log('\n=== 新三国杀道具定价表.xlsx (道具定价表) ===');
console.log('列名:', Object.keys(prices[0] || {}));
console.log('数据量:', prices.length);
console.log('示例数据:', JSON.stringify(prices[0], null, 2));

// 保存完整数据用于分析
console.log('\n' + '='.repeat(60));
console.log('保存完整数据到 temp_excel_data.json 供进一步分析...');
console.log('='.repeat(60));

const tempData = {
    generals: { columns: Object.keys(generals[0] || {}), count: generals.length, sample: generals.slice(0, 5) },
    items: { columns: Object.keys(items[0] || {}), count: items.length, sample: items.slice(0, 5) },
    prices: { columns: Object.keys(prices[0] || {}), count: prices.length, sample: prices.slice(0, 10) }
};

fs.writeFileSync(path.join(dataDir, 'temp_excel_data.json'), JSON.stringify(tempData, null, 2), 'utf8');
console.log('已保存到 temp_excel_data.json');
